kore/src/Kore/Step/Rule/Simplify.hs

```diff
@@ -0,0 +1,74 @@
+{-|
+Copyright   : (c) Runtime Verification, 2019
+License     : NCSA
+-}
+
+module Kore.Step.Rule.Simplify
+    ( simplifyOnePathRuleLhs
+    ) where
+
+import qualified Branch as BranchT
+    ( gather
+    , scatter
+    )
+import Kore.Internal.Conditional
+    ( Conditional (Conditional)
+    )
+import qualified Kore.Internal.Conditional as Conditional.DoNotUse
+    ( Conditional (..)
+    )
+import qualified Kore.Internal.Conditional as Conditional
+    ( withCondition
+    )
+import qualified Kore.Internal.Predicate as Predicate
+    ( fromPredicate
+    )
+import Kore.Step.Rule
+    ( OnePathRule (..)
+    , RulePattern (RulePattern)
+    )
+import qualified Kore.Step.Rule as RulePattern
+    ( RulePattern (..)
+    , fullyApplySubstitution
+    )
+import qualified Kore.Step.Simplification.Pattern as Pattern
+    ( simplifyAndRemoveTopExists
+    )
+import Kore.Step.Simplification.Simplify
+    ( MonadSimplify
+    , SimplifierVariable
+    )
+
+simplifyOnePathRuleLhs
+    ::  ( MonadSimplify simplifier
+        , SimplifierVariable variable
+        )
+    => OnePathRule variable
+    -> simplifier [OnePathRule variable]
```

Comment:
Would it make sense to return a `MultiAnd` here instead of a list? I'm thinking that because a disjunction on the left-hand side obligates us to prove all the claims, and also because there might be duplicated claims in this list.

Reply:
MultiAnd requires a `TopBottom` instance. I did that, but I'm not sure that the result is better.